src/fenwick.jl

```diff
@@ -0,0 +1,96 @@
+"""
+    FenwickTree{T}(n)
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) of length `n`.
+ 
+"""
+mutable struct FenwickTree{T}
+    bit::Vector{T}
+    n::Integer
+end
+
+FenwickTree{T}() where T = FenwickTree{T}(0)
+FenwickTree{T}(n::Integer) where T = FenwickTree{T}(zeros(T, n), n)
+
+"""
+    FenwickTree(arr::AbstractArray) 
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) from `arr`.
+ 
+"""
+function FenwickTree(a::AbstractVector{U}) where U
+    n = length(a)
+    bit = FenwickTree{U}(n)
+    @inbounds for i = 1:n
+        update!(bit, i, a[i])
+    end
+    bit
+end
+
+bit(F::FenwickTree{T}) where T = F.bit
+length(F::FenwickTree{T}) where T = F.n
+
+"""
+    update!(F::FenwickTree{T}, ind, val::T)
+
+Update the value of the [`FenwickTree`] by `val` from the index `ind` upto the length of the Fenwick Tree.
+
+""" 
+function update!(F::FenwickTree{T}, ind::Int, val::T) where T
+    i = ind
+    n = F.n
+    @boundscheck 1 <= i <= n || throw(ArgumentError("$i should be in between 1 and $n"))
+    @inbounds while i <= n
+        F.bit[i] += val
+        i += i&(-i)
+    end
+end
+
+"""
+    update!(F::FenwickTree{T}, left, right, val::T)
+
+Update the value of the [`FenwickTree`] by `val` from the index `left` upto the index `right`.
+
+"""    
+function update!(F::FenwickTree{T}, left::Int, right::Int, val::T) where T
+    update!(F, left, +val)
+    if (right+1 <= length(F))
+        update!(F, right+1, -val)
+    end
+end
+
+"""
+    sum(F::FenwickTree{T}, ind)
+    
+Return the cumulative sum from index 1 upto `ind` of the [`FenwickTree`](@ref)
+
+# Examples
+```
+julia> f = FenwickTree{Int}(6)
+julia> update!(f, 2, 5)
+julia> sum(f, 1)
+ 0
+julia> sum(f, 3)
+ 5
+julia> update!(f, 1, 4, 3)
+julia> sum(f, 1)
+ 3
+julia> sum(f, 3)
+ 8
+julia> sum(f, 6)
+ 5
+```
+"""
+function sum(F::FenwickTree{T}, ind::Int) where T
```

Comment:
I agree with @SyxP's comment that `sum` is not the best name for this function.  

`cumsum` is a Julia function which calculates the cumulative sum, but it returns an array of all values up to the end of an array. 

I would suggest `prefixsum()`, which is used in the description you linked to on Wikipedia.